pkg/backends/calico/routes.go

```diff
@@ -195,24 +197,104 @@ func (rg *routeGenerator) setRouteForSvc(svc *v1.Service, ep *v1.Endpoints) {
 	// see if any endpoints are on this node and advertise if so
 	// else remove the route if it also already exists
 	rg.Lock()
-	if rg.advertiseThisService(svc, ep) {
+	defer rg.Unlock()
+
+	advertise := rg.advertiseThisService(svc, ep)
+	if advertise {
 		route := svc.Spec.ClusterIP + "/32"
-		if cur, exists := rg.svcRouteMap[key]; !exists {
+		if cur, exists := rg.svcClusterRouteMap[key]; !exists {
 			// This is a new route - send it through.
-			rg.advertiseRoute(key, route)
+			rg.advertiseClusterRoute(key, route)
 		} else if route != cur {
 			// The route has changed somehow. Send a delete for the old one
 			// and add the new one. We don't expect to hit this, since cluster IPs
 			// are immutable. But, handle it for good measure.
 			logc.Warn("ClusterIP for service changed, adjusting")
-			rg.withdrawRoute(key, cur)
-			rg.advertiseRoute(key, route)
+			rg.withdrawClusterRoute(key, cur)
+			rg.advertiseClusterRoute(key, route)
 		}
-	} else if cur, exists := rg.svcRouteMap[key]; exists {
+	} else if cur, exists := rg.svcClusterRouteMap[key]; exists {
 		// We were advertising this route, but should no longer do so.
-		rg.withdrawRoute(key, cur)
+		rg.withdrawClusterRoute(key, cur)
+	}
+
+	// External IP's
+	if advertise {
+
```

Comment:
Just a stylistic difference, but we generally try to minimize the number of blank lines before / after `if` statements.

See [this line](https://github.com/projectcalico/confd/pull/266/files#diff-4d8eb9d6aa9442f0a7b3da984d5bd9c9R209) and [this line](https://github.com/projectcalico/confd/pull/266/files#diff-4d8eb9d6aa9442f0a7b3da984d5bd9c9R148) as examples.